src/pipelinedefinition.cpp

```diff
@@ -777,7 +786,34 @@ Status PipelineDefinition::getOutputsInfo(tensor_map_t& outputsInfo, const Model
 
                 for (const auto& [alias, realName] : specificDependencyMapping) {
                     const auto& finalName = dependencyNodeInfo->outputNameAliases.count(alias) > 0 ? dependencyNodeInfo->outputNameAliases.at(alias) : alias;
-                    outputsInfo[realName] = instance->getOutputsInfo().at(finalName);
+                    auto tensorInfo = instance->getOutputsInfo().at(finalName);
+                    if (gatherShape.size() > 0) {
+                        outputsInfo[realName] = tensorInfo->createCopyWithAdditionalShape(gatherShape);
+                    } else {
+                        outputsInfo[realName] = tensorInfo;
+                    }
+                }
+                break;
+            }
+            case NodeKind::CUSTOM: {
+                if (!dependencyNodeInfo->library.isValid()) {
+                    return StatusCode::NODE_LIBRARY_MISSING;
+                }
+
+                tensor_map_t info;
+                auto status = this->getCustomNodeMetadata(*dependencyNodeInfo, info, dependencyNodeInfo->library.getOutputsInfo);
+                if (!status.ok()) {
+                    return status;
+                }
+
+                for (const auto& [alias, realName] : specificDependencyMapping) {
+                    const auto& finalName = dependencyNodeInfo->outputNameAliases.count(alias) > 0 ? dependencyNodeInfo->outputNameAliases.at(alias) : alias;
+                    auto tensorInfo = info.at(finalName);
+                    if (gatherShape.size() > 0) {
+                        outputsInfo[realName] = tensorInfo->createCopyWithAdditionalShape(gatherShape);
+                    } else {
+                        outputsInfo[realName] = tensorInfo;
+                    }
```

Comment:
extract:
1) getDLNodeOutputInfo()
2) getCustomNodeOutputInfo()
3) Comented lines as getFinalShape (name TBD) since this is used for both nodes.

Reply:
I think this is already separated with a switch. I don't see a reason for it to be a method

Reply:
Done